app/api/v2/resource/addresses.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module API::V2
+  module Resource
+    # Addresses API
+    class Addresses < Grape::API
+      desc 'Documents related routes'
+      resource :addresses do
+        desc 'Upload a new address approval document for current user',
+             security: [{ 'BearerToken': [] }],
+             success: { code: 201, message: 'Document is uploaded' },
+             failure: [
+               { code: 400, message: 'Required params are empty' },
+               { code: 401, message: 'Invalid bearer token' },
+               { code: 422, message: 'Validation errors' }
+             ]
+        params do
+          requires :country,
+                   type: String,
+                   allow_blank: false,
+                   desc: 'Document type'
+          requires :address,
+                   type: String,
+                   allow_blank: false,
+                   desc: 'Document number'
+          requires :upload,
+                   desc: 'Array of Rack::Multipart::UploadedFile'
+          requires :city,
+                   allow_blank: true,
+                   desc: 'Document expiration date'
+          requires :postcode, type: String, desc: 'Any additional key: value pairs in json string format'
+        end
+
+        post do
+          identificator = SecureRandom.hex(16)
+
+          params[:upload].each do |file|
+            doc = current_user.documents.new(upload: file, identificator: identificator, doc_type: 'Address')
+
+            code_error!(doc.errors.details, 422) unless doc.save
+          end
+
+          KycService.address_step(
```

Comment:
we start address_step from controller, because we still use `Document` model both for address and documents.
And, what is more important, some of the info is custom and cannot be stored in document model. (city, postcode).

Maybe we can move this params to metadata and make a switch in model.